src/ReaderScreen/components/AudioPlayer/hooks/useAudioManifest.js

```diff
@@ -0,0 +1,155 @@
+import { useState, useEffect } from "react";
+import { DocumentDirectoryPath } from "react-native-fs";
+import { useSelector, useDispatch } from "react-redux";
+import { actions, logError } from "@common";
+import { fetchManifest } from "@service";
+
+const useAudioManifest = (baniID) => {
+  const [tracks, setTracks] = useState([]);
+  const [currentPlaying, setCurrentPlaying] = useState(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const defaultAudio = useSelector((state) => state.defaultAudio);
+
+  const dispatch = useDispatch();
+  const audioManifest = useSelector((state) => state.audioManifest);
+
+  const fetchAudioManifest = async () => {
```

Comment:
Currently this function is handling multiple jobs, can we break it down to multiple more focussed functions
- `mapApiDataToTracks`
- `mergeDownloadedTracks`
- `setDefaultTrack`

Reply:
done